src/legacy/ui/public/autoload/settings.js

```diff
@@ -28,7 +28,8 @@ import moment from 'moment-timezone';
 import chrome from '../chrome';
 
 function setDefaultTimezone(tz) {
-  moment.tz.setDefault(tz);
+  const timezone = tz === 'Browser' ? moment.tz.guess() : tz;
+  moment.tz.setDefault(timezone);
```

Comment:
@joshdover I seem to recall you were porting this to NP? You ok with this change?